### PR TITLE
Fix formatting of types with ocp-indent-compat=true

### PR DESCRIPTION
**Pull request by gpetiot**
This fix is needed for PR #524 (error on js_source.ml with `ocp-indent-compat=true`).

### CODE REVIEW

File: test/passing/ocp_indent_compat.ml [hunk 1]
```diff
@@ -2,3 +2,10 @@
 
 (* Bad: unboxing the function type *)
 external i : (int -> float[@unboxed]) = "i" "i_nat"
+
+module type M = sig
+  val doc_atrs :
+        (string Location.loc * payload) list
```
Comment (hhugo) [inline test/passing/ocp_indent_compat.ml]: I suspect this does not work when piped to ocp-indent

Comment (gpetiot) [reply test/passing/ocp_indent_compat.ml]: I'm working on writing the colon on the line below

**Comment by gpetiot** [comment]
I improved the formatting when `ocp-indent-compat=true` but when running `test_branch.sh` we can see minor regressions such as:
```
  val imported_sets_of_closures_table
    :
    Simple_value_approx.function_declarations option
    Set_of_closures_id.Tbl.t
```
Is this really a deal-breaker? Having different box structures depending on options and AST makes it quite tricky to get the right output.

**Comment by gpetiot** [comment]
I'm not sure I'm heading in the right direction with this PR, I will continue working on it but new ideas are welcomed.

**Comment by gpetiot** [comment]
No changes with `test_branch.sh fix-ocp-indent-compat-types 'ocp-indent-compat=false' 'ocp-indent-compat=false'`.
No regressions with `test_branch.sh fix-ocp-indent-compat-types 'ocp-indent-compat=true' 'ocp-indent-compat=true'`, trailing space after identificators are removed.

**Comment by gpetiot** [comment]
Yes I didn't found an alternative, the slightest change used to cause a lot of regressions, I struggled a lot on this PR.
Agreed on the parameter name.

**Pull request merged**